### PR TITLE
add ci

### PR DESCRIPTION
**Pull request by zunda-arrow**
Add CI.
Adds:
	- Codespell
	- black
	- ruff

Enabled `RUF006` since i find dangling tasks are a really easy error to make on this type of project.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: ci
+on: [push, pull_request]
+
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
```
Comment (circuitsacul) [inline .github/workflows/ci.yml]: These should be v3 I believe

File: noxfile.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    session.run("ruff", "check", *SRC_FILES)
+    session.run("codespell", *SRC_FILES)
+
+
+@nox.session
+def fix(session: nox.Session) -> None:
+    install_poetry_groups(session, "linting")
+    session.run("black", *SRC_FILES)
+    session.run("ruff", "check", "--fix", *SRC_FILES)
+    session.run("codespell", "-w", "-i2", *SRC_FILES)
```
Comment (circuitsacul) [inline noxfile.py]: Does "." not work? That's what I've always used.

**Review by circuitsacul — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (zunda-arrow) [reply .github/workflows/ci.yml]: changed it to v4

File: noxfile.py (hunk 2 above)
Comment (zunda-arrow) [reply noxfile.py]: changed to `.`

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: ci
+on: [push, pull_request]
+
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (circuitsacul) [inline .github/workflows/ci.yml]: @v3 is the latest

https://github.com/actions/checkout

**Review by circuitsacul — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (zunda-arrow) [reply .github/workflows/ci.yml]: done

**Review by circuitsacul — APPROVED** [review]

**Pull request merged**